File: LibraryManagement/src/app/models/borrow-history.model.ts
import { BookChapterStatus } from "../enums/book-chapter-status";
import { BorrowHistoryStatus } from "../enums/borrow-history-status";
import { IBook } from "./book.model";
import { IBookChapter } from "./bookchapter.model";
import { ILibraryCardInfo } from "./library-card.model";

export interface IBorrowHistoryInfo {
    id: string;
    borrowDate: string;
    endDate: string;
    status: BorrowHistoryStatus;
    bookChapterId: string;
    libraryCardId: string;
    bookChapter?: IBookChapter;
    libraryCard?: ILibraryCardInfo;
    bookId?: string;
    book?: IBook;
    lostOrDestroyedDate?: string;
}

export interface IEditRecordInfo {
    id: string;
    borrowDate: string;
    endDate: string;
    bookChapterId: string;
    libraryCardId: string;
    status: BorrowHistoryStatus;
}

File: LibraryAPI/ViewModels/BorrowHistory/BorrowHistoryModel.cs
using LibraryAPI.ViewModels.Book;
using LibraryAPI.ViewModels.LibraryCard;

namespace LibraryAPI.ViewModels.BorrowHistory
{
    public class BorrowHistoryModel
    {
        public Guid? Id { get; set; }

        public DateTime? BorrowDate { get; set; }

        public DateTime? EndDate { get; set; }

        public int? Status { get; set; }

        public Guid? BookChapterId { get; set; }

        public Guid? LibraryCardId { get; set; }
        public string? BookName { get; set; }
        public int? Chapter { get; set; }

        public virtual BookChapterModel? BookChapter { get; set; }
        public virtual LibraryCardBasicInfoModel? LibraryCard { get; set; }
    }
}
